flow360/component/simulation/validation/validation_simulation_params.py

```diff
@@ -39,6 +50,71 @@ def _check_consistency_wall_function_and_surface_output(v):
     return v
 
 
+def _check_cht_solver_settings(params):
+    has_heat_transfer = False
+
+    models = params.models
+
+    if models:
+        for model in models:
+            if isinstance(model, Solid):
+                has_heat_transfer = True
+
+        if has_heat_transfer is False:
+            params = _validate_cht_no_heat_transfer(params)
+        if has_heat_transfer is True:
+            params = _validate_cht_has_heat_transfer(params)
+
+    return params
+
+
+def _validate_cht_no_heat_transfer(params):
+
+    if params.outputs:
+        for output in params.outputs:
+            if isinstance(
+                output, (SurfaceOutput, VolumeOutput, SliceOutput, ProbeOutput, IsosurfaceOutput)
+            ):
+                if "residualHeatSolver" in output.output_fields.items:
+                    raise ValueError(
+                        f"Heat equation output variables: residualHeatSolver is requested in {output.output_type} with"
+                        " no definition of solid models."
```

Comment:
```suggestion
                        " no `Solid` model defined."
```

Reply:
Change other "solid" like this too.

Reply:
Yes, all the other solid model are changed according to the comment.